core/include/traccc/edm/track_candidate_collection.hpp

```diff
@@ -0,0 +1,166 @@
+    ///
+    /// @return A (non-const) vector of scalar values
+    ///
+    TRACCC_HOST_DEVICE
+    auto& ndf() { return BASE::template get<1>(); }
+    /// The number of degrees of freedom of the track fit (const)
+    ///
+    /// @return A (const) vector of scalar values
+    ///
+    TRACCC_HOST_DEVICE
+    const auto& ndf() const { return BASE::template get<1>(); }
+
+    /// The chi square of the track fit (non-const)
+    ///
+    /// @return A (non-const) vector of scalar values
+    ///
+    TRACCC_HOST_DEVICE
+    auto& chi2() { return BASE::template get<2>(); }
+    /// The chi square of the track fit (const)
+    ///
+    /// @return A (const) vector of scalar values
+    ///
+    TRACCC_HOST_DEVICE
+    const auto& chi2() const { return BASE::template get<2>(); }
+
+    /// The p-value of the track fit (non-const)
+    ///
+    /// @return A (non-const) vector of scalar values
+    ///
+    TRACCC_HOST_DEVICE
+    auto& pval() { return BASE::template get<3>(); }
+    /// The p-value of the track fit (const)
+    ///
+    /// @return A (const) vector of scalar values
+    ///
+    TRACCC_HOST_DEVICE
+    const auto& pval() const { return BASE::template get<3>(); }
+
+    /// The number of holes in the track pattern (non-const)
+    ///
+    /// @return A (non-const) vector of unsigned integers
+    ///
+    TRACCC_HOST_DEVICE
+    auto& nholes() { return BASE::template get<4>(); }
+    /// The number of holes in the track pattern (const)
+    ///
+    /// @return A (const) vector of unsigned integers
+    ///
+    TRACCC_HOST_DEVICE
+    const auto& nholes() const { return BASE::template get<4>(); }
+
+    /// The indices of the measurements associated to the track (non-const)
+    ///
+    /// @return A (non-const) jagged vector of unsigned integers
+    ///
+    TRACCC_HOST_DEVICE
+    auto& measurement_indices() { return BASE::template get<5>(); }
+    /// The indices of the measurements associated to the track (const)
+    ///
+    /// @return A (const) jagged vector of unsigned integers
+    ///
+    TRACCC_HOST_DEVICE
+    const auto& measurement_indices() const { return BASE::template get<5>(); }
+
+    /// @}
+
+    /// @name Utility functions
+    /// @{
+
+    /// Equality operator
+    ///
+    /// @note This function must only be used on proxy objects, not on
+    ///       containers!
+    ///
+    /// @param[in] other The object to compare with
+    /// @return @c true if the objects are equal, @c false otherwise
+    ///
+    template <typename T>
+    TRACCC_HOST_DEVICE bool operator==(const track_candidate<T>& other) const;
+
+    /// @}
+
+};  // class track_candidate
+
+/// SoA container describing reconstructed track candidates
+///
+/// @tparam ALGEBRA The algebra type used to describe the tracks
+///
+template <detray::concepts::algebra ALGEBRA>
+using track_candidate_collection = vecmem::edm::container<
+    track_candidate,
+    // params
+    vecmem::edm::type::vector<bound_track_parameters<ALGEBRA>>,
```

Comment:
@niermann999, this PR makes it once again a question how we want to deal with algebras in the long term. :thinking: Since our data model depends on the algebra that we use. :thinking: 

For now I did it like this. While using `traccc::default_algebra` everywhere for now to select the algebra. But I'm not in love with this setup...